### DISCUSSION TITLE
How to take array of strings as a input in move function?

### DISCUSSION THREAD

**Discussion by vivekratr** [aptos-labs/aptos-developer-discussions#255] [Questions] [answered]
### Describe your question in detail.

How to take array of strings as a input in move function?

here is the code: ```https://github.com/vivekratr/socialMove1```

### What have you tried or looked at?  Or how can we reproduce the error?

I have tried vector<vector<u8>> but not taking plain string as input

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by kshitijc1506** [comment]
Here's an example of a function taking in a vector (array) of strings and modifying each one to "hello". 

```
public fun take_string_arr(strings: &mut vector<String>) {
    let length = vector::length(strings);
    let i = 0;
    while (i < length) {
        let hello_str = string::utf8(b"hello");
        let element = vector::borrow_mut(strings, i);
        *element = hello_str; 
        i = i + 1;
    };
}
```

**Reply by 0xbe1** [reply]
i think the author is asking about how to pass in string array in Aptos CLI?

**Comment by 0xjunha** [comment, marked as answer]
I'm not sure if you're asking about passing multiple string arguments or `vector<String>` type, so here're scenarios for both cases.
### example module
```move
module test::test {
    use std::string::String;
    use std::vector;
    struct Note has key {
        title: String,
        content: String,
    }

    public entry fun write_note(author: &signer, title: String, content: String) {
        let note = Note{ title, content };
        move_to(author, note);
    }

    public entry fun write_note_in_array(author: &signer, note: vector<String>) {
        let content = vector::pop_back(&mut note);
        let title = vector::pop_back(&mut note);
        let note = Note{ title, content };
        move_to(author, note);
    }
}
```


### 1. multiple `String` args
run
```
aptos move run \
	--function-id $MODULE_ADDRESS::test::write_note \
	--args \
		"string:Some Title" \
		"string:Some Content" \
	--profile some_profile	
```

### 2. `vector<String>` arg
run
```
aptos move run \
	--function-id $MODULE_ADDRESS::test::write_note_in_array \
	--args \
		'string:["Some Title", "Some Content"]'
	--profile some_profile
```
